android-sdk/src/main/java/com/mobilecoin/lib/AccountKey.java

```diff
@@ -450,13 +446,38 @@ public String getFogReportId() {
         return accountBuilder.build();
     }
 
+    @Nullable
+    private Uri getNormalizedFogReportUri() {
+        if (fogReportUri == null) {
+            return  null;
+        }
+        try {
+            return new FogUri(fogReportUri).getUri();
+        } catch (InvalidUriException exception) {
+            throw new IllegalStateException("Bug: the Uri was already verified");
+        }
+    }
+
+    public boolean equivalent(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        AccountKey that = (AccountKey) o;
+        return Objects.equals(fogReportId, that.fogReportId) &&
+                getNormalizedFogReportUri().equals(that.getNormalizedFogReportUri()) &&
+                Arrays.equals(fogAuthoritySpki, that.fogAuthoritySpki) &&
+                viewKey.equals(that.viewKey) &&
+                spendKey.equals(that.spendKey) &&
+                subAddressViewKey.equals(that.subAddressViewKey) &&
+                subAddressSpendKey.equals(that.subAddressSpendKey);
+    }
+
     @Override
```

Comment:
This isn't a blocking comment, but somthing to note for the future. It might be worth exploring using [AutoValue](https://github.com/google/auto/tree/master/value) for this class. AutoValue provides `equals` and `hashCode` for free, and we wouldn't need to worry about unit testing that these methods work. This would help reduce boilerplate and number of unit tests. 

Can discuss more offline and also in relation to my unit testing project.

Reply:
sounds good, let's discuss this offline